server/controllers/user.js

```diff
@@ -0,0 +1,153 @@
+// import modules
+import jwt from 'jsonwebtoken';
+import bcrypt from 'bcrypt-nodejs';
+import models from '../models';
+
+require('dotenv').config();
+
+// create reference to db model
+const User = models.User;
+
+// create variable to hold environmental variable SECRET
+const secret = process.env.SECRET;
+
+export default {
+
+  /**
+     * @description creates a new user controller
+     * @method
+     * @param {Object} req - Request object
+     * @param {Object} res - Response object
+     * @returns {Object} json - payload
+     */
+  signup(req, res) {
+    User
+      .find({ where: { email: req.body.email } })
+      .then((response) => {
+        if (response) {
+          return res.status(409).send({
+            message: 'User already exists.',
+          });
+        }
+        User
+          .create({
+            firstName: req.body.firstName,
+            lastName: req.body.lastName,
+            email: req.body.email,
+            password: req.body.password,
+            imageUrl: req.body.imageUrl,
+          })
+          .then((user) => {
+            const userData = { user };
+            const token = jwt.sign(userData, secret);
+            return res.status(201).send({
+              message: 'Registered successfully',
+              token,
+            });
+          });
+      })
+      .catch(error => res.status(400).send({ message: error.message }));
+  },
+
+  /**
+     * @description sign in user controller
+     * @method
+     * @param {Object} req - Request object
+     * @param {Object} res - Response object
+     * @returns {Object} json - payload
+     */
+  signin(req, res) {
+    User
+      .find({ where: { email: req.body.email } })
+      .then((user) => {
+        if (!user) {
+          return res.status(404).send({ message: 'User is not registered!.' });
+        }
+
+        if (bcrypt.compareSync(req.body.password, user.password)) {
+          const userData = { user };
+          const token = jwt.sign(userData, secret);
+          return res.status(200).send({
+            message: 'Logged in Successfully',
+            token,
+          });
+        }
+        return res.status(409).send({
+          message: 'Invalide username or password',
+        });
+      })
+      .catch(error => res.status(400).send({ message: error.message }));
+  },
+
+  /**
+     * @description retrive user controller
+     * @method
+     * @param {Object} req - Request object
+     * @param {Object} res - Response object
+     * @returns {Object} json - payload
+     */
+  retrieve(req, res) {
+    // query the db to check if user already exits or not
+    User
+      .findOne({
+        where: {
+          id: req.decoded.user.id
+        },
+        attributes: { exclude: ['createdAt', 'updatedAt'] },
+      })
+      .then((user) => {
+        if (!user) {
+          return res.status(404).send({ message: 'User not found' });
+        }
+        return res.status(200).send({ user });
```

Comment:
It appears that your user table contains user password. You should exclude the password field on line 96 as well as this would constitute a security concern. Any user will be able to get another user's password (even if it is hashed).